lib/core/src/chain_swap.rs

```diff
@@ -534,22 +541,26 @@ impl ChainSwapStateHandler {
                 match swap.refund_tx_id.clone() {
                     None => {
                         warn!("Chain Swap {id} is in an unrecoverable state: {swap_state:?}");
-                        match self.verify_user_lockup_tx(swap).await {
-                            Ok(_) => {
-                                warn!("Chain Swap {id} user lockup tx has been broadcast. Attempting refund.");
-                                let refund_tx_id = self.refund_outgoing_swap(swap).await?;
-                                info!("Broadcast refund tx for Chain Swap {id}. Tx id: {refund_tx_id}");
-                                self.update_swap_info(
-                                    id,
-                                    RefundPending,
-                                    None,
-                                    None,
-                                    None,
-                                    Some(&refund_tx_id),
-                                )
-                                .await?;
+                        match swap.user_lockup_tx_id {
```

Comment:
Just a comment for @dangeross, I've reverted the script history check here to avoid re-fetching. It occurs anyway when reconstructing the refund transaction (`utils::new_lbtc_refund_tx`) so there shouldn't be any problems there.